src/App.js

```diff
@@ -1,59 +1,46 @@
+import './App.scss';
+
+import React, { useEffect, useState } from 'react';
+
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
+import { Routes } from './Routes';
 import routerParams from '@redhat-cloud-services/frontend-components-utilities/files/RouterParams';
-import { connect } from 'react-redux';
-import { matchPath } from 'react-router-dom';
 
-import { Routes } from './Routes';
-import './App.scss';
+const App = (props) => {
+    const [auth, setAuth] = useState(false);
+    const appNavClick = {
+        overview(redirect) { insights.chrome.appNavClick({ id: 'overview', redirect  }); },
+        rules(redirect) { insights.chrome.appNavClick({ id: 'rules', redirect  }); },
+        topics(redirect) { insights.chrome.appNavClick({ id: 'topics', redirect }); }
+    };
 
-class App extends Component {
-    componentDidMount() {
+    useEffect(() => {
         insights.chrome.init();
+        insights.chrome.auth.getUser().then(() => setAuth(true));
         insights.chrome.identifyApp('insights');
-        this.appNav = insights.chrome.on('APP_NAVIGATION', event => {
+        const baseComponentUrl = props.location.pathname.split('/')[1];
+        const unregister = insights.chrome.on('APP_NAVIGATION', event => {
             if (event.domEvent) {
-                this.props.history.push(`/${event.navId}`);
-            }
-
-            if (location.pathname.indexOf(event.navId) === -1 &&
-                !matchPath(location.href, { path: `${document.baseURI}insights/${event.navId}` })
-            ) {
-                this.props.history.push(`/${event.navId}`);
+                props.history.push(`/${event.navId}`);
+                appNavClick[baseComponentUrl](true);
             }
         });
-    }
-
-    componentDidUpdate(prevProps) {
-        if (this.props !== prevProps) {
-            const baseComponentUrl = location.pathname.slice(location.pathname.indexOf('insights/')).split('/')[1];
-            const appNavClick = {
-                topics() { insights.chrome.appNavClick({ id: 'topics' }); },
-                rules() { insights.chrome.appNavClick({ id: 'rules' }); }
-            };
-            if (appNavClick[baseComponentUrl]) {
-                appNavClick[baseComponentUrl]();
-            }
-        }
-    }
 
-    componentWillUnmount() {
-        this.appNav();
-    }
+        return () => unregister();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
-    render() {
-        return (
-            <Routes childProps={this.props} />
-        );
-    }
-}
+    useEffect(() => {
+        const baseComponentUrl = props.location.pathname.split('/')[1];
+        appNavClick[baseComponentUrl](false);
+    });
```

Comment:
I don't like this one. Can you update it to `useEffect` with some props watch? Like, when something from router updates? Because you are running this on every update and it can lead to infinit loop.

Reply:
OOOO yeaaaaaah lets add both appnavclick and props.location to that... 🤔  testing now...